### PR TITLE
Refactor API Gateway call to remove pem package

### LINKED ISSUE #182: Investigate whether to use `rejectUnauthorize` option in ClaimTracker's call to the API gateway.
### Description

Currently we are making use of `https.Agent` to call the API gateway using TLS certificates. When an Agent instance is created, it is passed an `options` object for handling the connection/TLS handshake.  This options object is where certificates are loaded, for instance.  We may want to implement additional security by implementing a `checkServerIdentity` function - [ see node http docs here](https://nodejs.org/api/https.html#https_https_request_url_options_callback).  

Since we are using self-signed certificates, the implementation might be a simple check that the ca and certificate are the same.  Or it is possible that we may not need to implement this function. 

[Context in this PR](https://github.com/cagov/ui-claim-tracker/pull/153/files#diff-48105795200c87e0b4fdf4423588e96c9978cb47303200b5d3678efd8b096df2R91)

[Somewhat relevant blog post](https://sebtrif.xyz/blog/2019-10-03-client-side-ssl-in-node-js-with-fetch/)

### Acceptance Criteria

  - [ ] Investigate whether to implement a `checkServerIdentity` function for our Agent instance we use to call API Gateway.
  - [ ] If necessary, create a ticket to implement a `checkServerIdentity` function.

### PR DESCRIPTION
**Pull request by rocketnova**
## Ticket

Resolves #182

## Changes

- Removes `pem` package
- Removes `getCertificate()` function and instead passes the PKCS#12 file directly to `https.Agent`
- Removes explicit setting of `rejectUnauthorized` and `keepAlive` and instead uses default values

## Context

To make the mTLS request to the API Gateway, we create a custom `https.Agent` and pass the agent as an option to [node-fetch](https://github.com/node-fetch/node-fetch) (other [options](https://blog.logrocket.com/5-ways-to-make-http-requests-in-node-js/) would also have been fine). 

Documented `https.Agent` options are in the `https.Agent` [new Agent](https://nodejs.org/api/https.html#https_new_agent_options) and [`http.Agent`](https://nodejs.org/api/http.html#http_class_http_agent) docs. However, it [turns out](https://github.com/node-fetch/node-fetch/issues/904#issuecomment-747828286) that `https.Agent` also accepts options for [`tls.connect()`](https://nodejs.org/api/tls.html#tls_tls_connect_port_host_options_callback), which in turn accepts options for [`tls.createSecureContext()`](https://nodejs.org/api/tls.html#tls_tls_createsecurecontext_options). 

`tls.createSecureContext()` accepts `pfx` as an option, so this PR removes the `pem` package and passes the pfx file directly to the `https.Agent` instead of [reading it](https://dexus.github.io/pem/jsdoc/module-pem.html#.readPkcs12) into separate cert and key files. This simplifies the code and reduces dependencies.

This PR uses the default value (false) for the `http.Agent` option `keepAlive` because we are only making one connection, so it is unnecessary to keep the connection open.

We use the default value (true) for the `tls.connect()` option `rejectUnauthorized` because we only want to accept TLS certs _from_ the API gateway that have been signed by known and trusted CAs.

We also do not use the `tls.connect()` option `checkServerIdentity` because we want to use the built-in `tls.checkServerIdentity()` for verify the server identity. 

## Testing Instructions

To test this is working as expected:

0. [Set up `.env`](https://github.com/cagov/ui-claim-tracker#environment-variables)
1. Fetch remote branches: `git fetch`
2. Checkout this branch: `git checkout rocket/182-mtls`
3. Update your local packages: `yarn install`
4. Run the app locally: `yarn dev`
5. Navigate to the app in your browser: `localhost:3000/claimstatus`
6. Verify in either the browser console or local terminal window where you ran `yarn dev` that the result of the call to the API gateway is `{ claimData: [ { ClaimType: null } ] }`

### CODE REVIEW

**Review by lomky — COMMENTED** [review]
One nice to have inline reference comment swap - commenting for now to check on what testing to do!

File: pages/index.tsx [hunk 1]
```diff
@@ -80,34 +78,15 @@ export const getServerSideProps: GetServerSideProps = async ({ req, locale }) =>
...
-    // @ts-ignore -- TypeScript does not handle promisify well.
-    const keybundle = await pemReadPkcs12(pfx)
-    return keybundle
-  }
-
-  // Takes certificate that getCertificate function returns as argument,
-  // makes API call, returns all API data.
-  async function makeRequest(certificate: Pkcs12ReadResult) {
+  // Makes API call, returns all API data.
+  async function makeRequest() {
```
[the remark this answers is not in the document]
Comment (rocketnova) [reply pages/index.tsx:82]: Yup. I think that Kevin kept everything in `index.tsx`, but I think we could re-arrange where things live. Where are you thinking?

File: pages/index.tsx [hunk 2]
```diff
@@ -80,34 +78,15 @@ export const getServerSideProps: GetServerSideProps = async ({ req, locale }) =>
...
     // https://nodejs.org/api/tls.html#tls_tls_createsecurecontext_options
     const options = {
-      cert: certificate.cert,
-      key: certificate.key,
-      rejectUnauthorized: false,
-      keepAlive: false,
+      pfx: fs.readFileSync(P12_PATH),
     }
 
     // Instantiate agent to use with TLS Certificate.
```
[the remark this answers is not in the document]
Comment (rocketnova) [reply pages/index.tsx:92]: Good call. Done!

File: pages/index.tsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (lomky) [reply pages/index.tsx:82]: Given our small amount of expected logic code, I was thinking just a singular helper/util file to start with. If we outgrow it we can think more about a structured layout.

[the remark this answers is not in the document]
Comment (rocketnova) [reply pages/index.tsx:82]: Created #221

**Review by lomky — APPROVED** [review]
Nice! 🎉 :shipit:

**Pull request merged**